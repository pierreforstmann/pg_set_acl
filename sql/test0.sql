drop extension if exists pg_set_acl;
--
create extension pg_set_acl;
--
create user test;
--
select pgsa_grant('work_mem','test');
select * from pg_set_acl;
select pgsa_revoke('work_mem','test');
--
drop extension pg_set_acl;
