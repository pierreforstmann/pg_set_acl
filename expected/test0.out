--
create user postgres superuser;
--
\c contrib_regression postgres
drop extension if exists pg_set_acl;
NOTICE:  extension "pg_set_acl" does not exist, skipping
--
create extension pg_set_acl;
--
select set_acl.grant('s','u');
ERROR:  Cannot find setting s
select set_acl.grant('work_mem','u');
ERROR:  Cannot find user u
--
create user test;
--
\c contrib_regression test
set work_mem='1GB';
ERROR:  pg_set_acl: permission denied for (work_mem,test)
--
\c contrib_regression postgres 
select set_acl.grant('work_mem','test');
 grant 
-------
 t
(1 row)

select * from set_acl.privs;
 privilege | parameter_name | user_name 
-----------+----------------+-----------
 SET       | work_mem       | test
(1 row)

--
\c contrib_regression test
set work_mem='1GB';
show work_mem;
 work_mem 
----------
 1GB
(1 row)

select set_config('work_mem', '500MB', false);
ERROR:  permission denied for function set_config
show work_mem;
 work_mem 
----------
 1GB
(1 row)

--
\c contrib_regression postgres
select set_acl.revoke('work_mem','test');
 revoke 
--------
 t
(1 row)

--
\c contrib_regression test
set work_mem='1GB';
ERROR:  pg_set_acl: permission denied for (work_mem,test)
select set_config('work_mem', '500MB', false);
ERROR:  permission denied for function set_config
show work_mem;
 work_mem 
----------
 4MB
(1 row)

--
\c contrib_regression postgres
drop extension pg_set_acl;
--
